src/v2/models/application.js

```diff
@@ -863,7 +863,13 @@ export default class ApplicationModel extends GenericModel {
       throw err;
     });
 
-    return routes.filter((route) => _.get(route, 'metadata.labels["app.kubernetes.io/part-of"]') === 'argocd' && _.get(route, 'spec.host'))[0];
+    if (routes && routes.length) {
+      // route exists
+      const routeObj = routes.filter((route) => _.get(route, 'metadata.labels["app.kubernetes.io/part-of"]', '') === 'argocd');
+      return routeObj && routeObj.length ? routeObj[0] : '';
```

Comment:
It should return undefined for the route if not found
return routeObj.length > 0 ? routeObj[0] : undefined;